apis/tempo/v1alpha1/microservices_webhook.go

```diff
@@ -28,11 +31,10 @@ func (r *Microservices) SetupWebhookWithManager(mgr ctrl.Manager, defaultImages
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(r).
 		WithDefaulter(&defaulter{defaultImages: defaultImages}).
+		WithValidator(&validator{client: mgr.GetClient()}).
 		Complete()
 }
 
-// TODO(user): EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-
 //+kubebuilder:webhook:path=/mutate-tempo-grafana-com-v1alpha1-microservices,mutating=true,failurePolicy=fail,sideEffects=None,groups=tempo.grafana.com,resources=microservices,verbs=create;update,versions=v1alpha1,name=mmicroservices.kb.io,admissionReviewVersions=v1
 
 type defaulter struct {
```

Comment:
In this case I've followed the approach taken by opentelemetry-operator, which doesn't update the CR. In general I like to keep configurations minimal and tidy, and only specify/show non-default values. This might be a special case, because the default value contains the instance name, but I'd still consider it as a default value and therefore tend to not show it (i.e not set the value, and therefore it won't show up in the yaml/json). wdyt?

Reply:
note that the OTEL operator started without webhooks. 

A better question to ask here is whether users should know the SA name, if yes then showing it in the CR is beneficial. 

> his might be a special case, because the default value contains the instance name, but I'd still consider it as a default value and therefore tend to not show it (i.e not set the value, and therefore it won't show up in the yaml/json). wdyt?

I would prefer to keep the manifest files simple and set defaults in the CR - it's better to users  to understand what is the current configuration.